src/server/game/Spells/SpellEffects.cpp

```diff
@@ -5612,17 +5612,11 @@ void Spell::EffectCreateAreaTrigger(SpellEffIndex /*effIndex*/)
     if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT)
         return;
 
-    Position pos;
     if (!m_targets.HasDst())
-        pos = GetCaster()->GetPosition();
-    else
-        pos = destTarget->GetPosition();
-
-    // trigger entry/miscvalue relation is currently unknown, for now use MiscValue as trigger entry
-    uint32 triggerEntry = effectInfo->MiscValue;
+        return;
 
-    AreaTrigger * areaTrigger = new AreaTrigger;
-    if (!areaTrigger->CreateAreaTrigger(GetCaster()->GetMap()->GenerateLowGuid<HighGuid::AreaTrigger>(), triggerEntry, GetCaster(), GetSpellInfo(), pos, m_SpellVisual))
+    AreaTrigger* areaTrigger = new AreaTrigger();
+    if (!areaTrigger->CreateAreaTrigger(effectInfo->MiscValue, GetCaster(), nullptr, GetSpellInfo(), destTarget->GetPosition(), GetSpellInfo()->GetDuration(), m_SpellVisual, m_castId))
```

Comment:
Incorrect behavior due to SPELLMOD_DURATION.
Consecration's base duration (DurationEntry->Duration, which is returned by GetSpellInfo()->GetDuration()) is 9000 ms, but learning 3 ranks of http://www.wowhead.com/spell=209218/consecration-in-flame will extend it to 12000 ms. It will cause the periodic dummy to deal damage 12 times, but the visual of consecration fades after 9 seconds